sympy/geometry/line.py

```diff
@@ -1083,24 +1093,68 @@ class Line(LinearEntity):
     >>> s = Segment((0, 0), (0, 1))
     >>> Line(s).equation()
     x
+
+    >>> # a circle object is returned
+    >>> from sympy.abc import x, y, a, b
+    >>> Line(3 * x + y + 18)
+    Line2D(Point2D(0, 0), Point2D(1, -3))
+    >>> Line(3 * a + b + 18, x='a', y='b')
+    Line2D(Point2D(0, 0), Point2D(1, -3))
     """
 
-    def __new__(cls, p1, p2=None, **kwargs):
-        if isinstance(p1, LinearEntity):
-            if p2:
-                raise ValueError('If p1 is a LinearEntity, p2 must be None.')
-            dim = len(p1.p1)
-        else:
-            p1 = Point(p1)
-            dim = len(p1)
-            if p2 is not None or isinstance(p2, Point) and p2.ambient_dimension != dim:
-                p2 = Point(p2)
+    def __new__(cls, *args, **kwargs):
+
+        if len(args) == 1 and isinstance(args[0], Expr):
+            x = kwargs.get('x', 'x')
+            y = kwargs.get('y', 'y')
+            equation = args[0]
+
+            def find(x_, equation_):
```

Comment:
No need to obfuscate parameters. from `import this`, "Namespaces are one honking great idea -- let's do more of those!"

Reply:
@smichr Done. Please take a look.